rs/execution_environment/src/execution/update.rs

```diff
@@ -147,13 +147,31 @@ pub fn execute_update(
     let helper = match UpdateHelper::new(&clean_canister, &original, deallocation_sender) {
         Ok(helper) => helper,
         Err(err) => {
+            if err.code() == ErrorCode::CanisterWasmMemoryLimitExceeded
+                && original.call_or_task == CanisterCallOrTask::Task(CanisterTask::OnLowWasmMemory)
+            {
+                //`OnLowWasmMemoryHook` is taken from task_queue (i.e. `OnLowWasmMemoryHookStatus` is `Executed`),
+                // but its was not executed due to error `WasmMemoryLimitExceeded`. To ensure that the hook is executed
+                // when the error is resolved we need to set `OnLowWasmMemoryHookStatus` to `Ready`. Because of
+                // the way `OnLowWasmMemoryHookStatus::update` is implemented we first need to remove it from the
+                // task_queue (which calls `OnLowWasmMemoryHookStatus::update(false)`) followed with `enqueue`
+                // (which calls `OnLowWasmMemoryHookStatus::update(true)`) to ensure desired behavior.
+                clean_canister
+                    .system_state
+                    .task_queue
+                    .remove(ic_replicated_state::ExecutionTask::OnLowWasmMemory);
+                clean_canister
+                    .system_state
+                    .task_queue
+                    .enqueue(ic_replicated_state::ExecutionTask::OnLowWasmMemory);
```

Comment:
nit: since this is the second place we're doing this remove/enqueue operation, and we're adding many comments, should we move it into a separate function?

Reply:
That is a good point, though I am not sure. I also wanted to raise this point, but I am not sure if we should add such a function to the task queue since it will further diverge from the queue interface, it may be in the future mistaken with `enqueue` function.